scripts/lib/CIME/XML/generic_xml.py

```diff
@@ -82,84 +242,73 @@ def write(self, outfile=None):
             with open(outfile,'w') as xmlout:
                 xmlout.write(xmlstr)
 
-    def get_node(self, nodename, attributes=None, root=None, xpath=None):
+    def scan_child(self, nodename, attributes=None, root=None):
         """
         Get an xml element matching nodename with optional attributes.
 
         Error unless exactly one match.
         """
 
-        nodes = self.get_nodes(nodename, attributes=attributes, root=root, xpath=xpath)
+        nodes = self.scan_children(nodename, attributes=attributes, root=root)
 
         expect(len(nodes) == 1, "Incorrect number of matches, {:d}, for nodename '{}' and attrs '{}' in file '{}'".format(len(nodes), nodename, attributes, self.filename))
         return nodes[0]
 
-    def get_optional_node(self, nodename, attributes=None, root=None, xpath=None):
+    def scan_optional_child(self, nodename, attributes=None, root=None):
         """
         Get an xml element matching nodename with optional attributes.
 
         Return None if no match.
         """
-        nodes = self.get_nodes(nodename, attributes=attributes, root=root, xpath=xpath)
+        nodes = self.scan_children(nodename, attributes=attributes, root=root)
 
         expect(len(nodes) <= 1, "Multiple matches for nodename '{}' and attrs '{}' in file '{}'".format(nodename, attributes, self.filename))
         return nodes[0] if nodes else None
 
-    def get_nodes(self, nodename, attributes=None, root=None, xpath=None):
+    def scan_children(self, nodename, attributes=None, root=None):
 
-        logger.debug("(get_nodes) Input values: {}, {}, {}, {}, {}".format(self.__class__.__name__, nodename, attributes, root, xpath))
+        logger.debug("(get_nodes) Input values: {}, {}, {}, {}".format(self.__class__.__name__, nodename, attributes, root))
 
         if root is None:
             root = self.root
         nodes = []
 
-        expect(attributes is None or xpath is None,
-               " Arguments attributes and xpath are exclusive")
-        if xpath is None:
-            xpath = ".//"+nodename
+        xpath = ".//" + (nodename if nodename else "")
 
         if attributes:
             # xml.etree has limited support for xpath and does not allow more than
             # one attribute in an xpath query so we query seperately for each attribute
             # and create a result with the intersection of those lists
 
             for key, value in attributes.items():
-                if value is not None:
-                    expect(isinstance(value, six.string_types),
-                           " Bad value passed for key {}".format(key))
+                if value is None:
+                    xpath = ".//{}[@{}]".format(nodename, key)
```

Comment:
Add support for value=None, meaning we don't care what the value of the attribute is as long as the element has an attribute by that name.